cli-commands/etherlime-ide/etherlime-ide.js

```diff
@@ -0,0 +1,85 @@
+const fs = require('fs-extra');
+const path = require('path');
+const git = require('simple-git/promise')();
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const spawn = require('child_process').spawn;
+
+const ideRepoUrl = 'https://github.com/LimeChain/Solidity-IDE.git';
+const ideFolder = 'Solidity-IDE';
+const ideServerRun = 'Solidity-IDE/solc-server.js';
+
+let projectWorkingDir = process.cwd(); //save current working dir of the project
+
+const run = async (port) => {
+    let etherlimeRootDir = getRootDir();
+    await fetchIdeRepo(etherlimeRootDir);
+    await runIde(etherlimeRootDir, port)
+}
+
+// find etherlime root dir
+const getRootDir = () => {
+    let currentDir = path.parse(__dirname)
+    return path.dirname(currentDir.dir)
+}
+
+//clone IDE repo or pull new updates if it was already initialized
+const fetchIdeRepo = async (rootDir) => {
+
+    if(fs.existsSync(`${rootDir}/${ideFolder}`)) {
+        console.log("====== Updating IDE ======")
+        try {
+            changeCurrentWorkingDir(`${rootDir}/${ideFolder}`) //change working dir in order to pull the repo
+            await git.pull('origin', 'master')
+        } catch (e) {
+        }
+        
+        fs.removeSync('./node_modules');
+        await installIdeModules()
+        return
+    }
+
+    console.log('====== Initializing IDE ======')
+    await git.clone(ideRepoUrl, `${rootDir}/${ideFolder}`);
+    changeCurrentWorkingDir(`${rootDir}/${ideFolder}`); //change working dir to install the packages
+    await installIdeModules();
+}
+
+const installIdeModules = async () => {
+    console.log('====== Installing IDE modules ======')
+    await exec(`npm install`)
+}
+
+const runIde = async (rootDir, port) => {
+    console.log("====== Running IDE ======");
+    await exec('npm run build-local')
+    changeCurrentWorkingDir(projectWorkingDir); //return to project's working dir
+    runGanache(port) //run etherlime ganache
```

Comment:
What will happen if the developer has already ran etherlime ganache in another terminal? I think the assumption for all our commands (for example test and deploy) is that the developer runs their own ganache. I think we should keep not be running one for them this time too